StripePaymentSheet/StripePaymentSheet/Source/Internal/API Bindings/Link/ConsumerSession.swift

```diff
@@ -12,81 +12,50 @@ import UIKit
-              let dict = response["consumer_session"] as? [AnyHashable: Any],
-              let clientSecret = dict["client_secret"] as? String,
-              let emailAddress = dict["email_address"] as? String,
-              let redactedPhoneNumber = dict["redacted_phone_number"] as? String
-        else {
-            return nil
-        }
-
-        var verificationSessions = [VerificationSession]()
-        if let sessions = dict["verification_sessions"] as? [[AnyHashable: Any]] {
-            for session in sessions {
-                if let parsedSession = VerificationSession.decodedObject(fromAPIResponse: session) {
-                    verificationSessions.append(parsedSession)
-                }
-            }
-        }
-
-        let authSessionClientSecret = response["auth_session_client_secret"] as? String
-
-        let supportedPaymentDetailsTypeStrings = dict["support_payment_details_types"] as? [String]
-
-        let supportedPaymentDetailsTypes = supportedPaymentDetailsTypeStrings?.compactMap {
-            ConsumerPaymentDetails.DetailsType(rawValue: $0)
-        }
+    private enum CodingKeys: String, CodingKey {
+        case clientSecret
+        case emailAddress
+        case redactedPhoneNumber
+        case verificationSessions
+        case supportedPaymentDetailsTypes = "supportPaymentDetailsTypes"
+    }
 
-        return ConsumerSession(clientSecret: clientSecret,
-                               emailAddress: emailAddress,
-                               redactedPhoneNumber: redactedPhoneNumber,
-                               verificationSessions: verificationSessions,
-                               authSessionClientSecret: authSessionClientSecret,
-                               supportedPaymentDetailsTypes: supportedPaymentDetailsTypes,
-                               allResponseFields: dict) as? Self
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.clientSecret = try container.decode(String.self, forKey: .clientSecret)
+        self.emailAddress = try container.decode(String.self, forKey: .emailAddress)
+        self.redactedPhoneNumber = try container.decode(String.self, forKey: .redactedPhoneNumber)
+        self.verificationSessions = try container.decodeIfPresent([ConsumerSession.VerificationSession].self, forKey: .verificationSessions) ?? []
+        self.supportedPaymentDetailsTypes = try container.decodeIfPresent(Set<ConsumerPaymentDetails.DetailsType>.self, forKey: .supportedPaymentDetailsTypes) ?? []
     }
 
 }
 
-// MARK: - Cookie Management
-
-extension ConsumerSession {
-    func updateCookie(withStore store: LinkCookieStore) {
-        store.updateSessionCookie(with: authSessionClientSecret)
+extension ConsumerSession: Equatable {
+    static func ==(lhs: ConsumerSession, rhs: ConsumerSession) -> Bool {
+        // NSObject-style equality
+        return ObjectIdentifier(lhs) == ObjectIdentifier(rhs)
```

Comment:
Do we really want to compare the reference identifiers here? I'd expect this comparison to be based on the underlying data (or maybe the `allResponseFields`).

Reply:
I think that would be the superior approach, honestly - this was to preserve the behavior from when this type was an NSObject and we got identifier-based equality by default. Auditing all places we compare two session objects to ensure that content-based equality wouldn't break things felt out of scope, but I'm happy to add a followup ticket for it!